sympy/solvers/diophantine.py

```diff
@@ -1040,13 +1040,13 @@ def _diop_quadratic(var, coeff, t):
 
         if D < 0:
             for solution in solns_pell:
-                s1 = P*Matrix([solution[0], solution[1]]) + Q
-                s2 = P*Matrix([-solution[0], solution[1]]) + Q
-                try:
-                    sol.add(tuple([as_int(_) for _ in s1]))
-                    sol.add(tuple([as_int(_) for _ in s2]))
-                except ValueError:
-                    pass
+                for X_i in [-solution[0], solution[0]]:
```

Comment:
Not exactly,

What was happening before.
```python
In [1]: diophantine(x**2 + y**2 -5)
Matrix([[1], [2]])
Matrix([[-1], [2]])
Out[1]: set([(-1, 2), (1, 2)])

In [2]: diophantine(x**2 + y**2 -5,permute = True)
Matrix([[1], [2]])
Matrix([[-1], [2]])
Out[2]: set([(-1, -2), (-1, 2), (1, -2), (1, 2)])
```
Here matrix shown are the Matrix from which solution is extracted `Matrix([solution[0], solution[1]])` line :1042 in `diophantine.py`
Even after using `permute=True` this is just showing the solutions related to `(1, 2)` and `(2, 1)` is not considered 

While the modified code
```python
diophantine(x**2 + y**2 -5)
Matrix([[-2], [-1]])
Matrix([[-2], [1]])
Matrix([[2], [-1]])
Matrix([[2], [1]])
Matrix([[-1], [-2]])
Matrix([[-1], [2]])
Matrix([[1], [-2]])
Matrix([[1], [2]])
Out[1]: set([(-2, -1), (-2, 1), (-1, -2), (-1, 2), (1, -2), (1, 2), (2, -1), (2, 1)])
```
This is giving output related to all the permutations.